### PR TITLE
Keep X11 connection open for entire test run

### PR DESCRIPTION
**Pull request by dnnr**

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -488,13 +490,32 @@ def running_clients(hlwm, running_clients_num):
...
+    class X11Connection:
+        def __init__(self):
+            self.display = Xlib.display.Display()
+
+        def shutdown(self):
+            self.display.close()
+
+    x11_connection_ = X11Connection()
+    yield x11_connection_
+    x11_connection_.shutdown()
```
Comment (t-wissmann) [inline tests/conftest.py]: Here, we don't really need the class, right? We could just define:
```
display = Xlib.display.Display()
yield display
display.close()
```
But we can also keep the class for the case that we want to put more attributes here.

Comment (dnnr) [reply tests/conftest.py]: Oh, you're totally right! I was quite stuck in the pattern it seems... I'll change it

File: tests/conftest.py [hunk 2]
```diff
@@ -488,13 +490,25 @@ def running_clients(hlwm, running_clients_num):
...
 
+@pytest.fixture()
+def x11(x11_connection):
+    """ Short-lived fixture for interacting with the X11 display and creating
+    clients that are automatically destroyed at the end of each test. """
     class X11:
-        def __init__(self):
-            self.display = display.Display()
+        def __init__(self, x11_connection):
+            self.display = x11_connection.display
```
Comment (t-wissmann) [inline tests/conftest.py]: This now becomes `self.display = x11_connection`

Comment (dnnr) [reply tests/conftest.py]: Oops, thanks for fixing it

**Pull request merged**